Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -138,7 +144,11 @@ void TextureCacheBase::Cleanup(int _frameCount)
   TexAddrCache::iterator tcend = textures_by_address.end();
   while (iter != tcend)
   {
-    if (iter->second->frameCount == FRAMECOUNT_INVALID)
+    if (iter->second->tmem_only)
+    {
+      iter = InvalidateTexture(iter);
```

Comment:
Ah, right you are. Though it is more than a little weird to have a function called Invalidate texture that only invalidates sometimes.